Source/ThirdParty/xss.js

```diff
@@ -0,0 +1,1615 @@
+    caption: [],
+    center: [],
+    cite:   [],
+    code:   [],
+    col:    ['align', 'valign', 'span', 'width'],
+    colgroup: ['align', 'valign', 'span', 'width'],
+    dd:     [],
+    del:    ['datetime'],
+    details: ['open'],
+    div:    [],
+    dl:     [],
+    dt:     [],
+    em:     [],
+    font:   ['color', 'size', 'face'],
+    footer: [],
+    h1:     [],
+    h2:     [],
+    h3:     [],
+    h4:     [],
+    h5:     [],
+    h6:     [],
+    header: [],
+    hr:     [],
+    i:      [],
+    img:    ['src', 'alt', 'title', 'width', 'height'],
+    ins:    ['datetime'],
+    li:     [],
+    mark:   [],
+    nav:    [],
+    ol:     [],
+    p:      [],
+    pre:    [],
+    s:      [],
+    section:[],
+    small:  [],
+    span:   [],
+    sub:    [],
+    sup:    [],
+    strong: [],
+    table:  ['width', 'border', 'align', 'valign'],
+    tbody:  ['align', 'valign'],
+    td:     ['width', 'rowspan', 'colspan', 'align', 'valign'],
+    tfoot:  ['align', 'valign'],
+    th:     ['width', 'rowspan', 'colspan', 'align', 'valign'],
+    thead:  ['align', 'valign'],
+    tr:     ['rowspan', 'align', 'valign'],
+    tt:     [],
+    u:      [],
+    ul:     [],
+    video:  ['autoplay', 'controls', 'loop', 'preload', 'src', 'height', 'width']
+  };
+}
+
+// 默认CSS Filter
+var defaultCSSFilter = new FilterCSS();
+
+/**
+ * 匹配到标签时的处理方法
+ *
+ * @param {String} tag
+ * @param {String} html
+ * @param {Object} options
+ * @return {String}
+ */
+function onTag (tag, html, options) {
+  // do nothing
+}
+
+/**
+ * 匹配到不在白名单上的标签时的处理方法
+ *
+ * @param {String} tag
+ * @param {String} html
+ * @param {Object} options
+ * @return {String}
+ */
+function onIgnoreTag (tag, html, options) {
+  // do nothing
+}
+
+/**
+ * 匹配到标签属性时的处理方法
+ *
+ * @param {String} tag
+ * @param {String} name
+ * @param {String} value
+ * @return {String}
+ */
+function onTagAttr (tag, name, value) {
+  // do nothing
+}
+
+/**
+ * 匹配到不在白名单上的标签属性时的处理方法
+ *
+ * @param {String} tag
+ * @param {String} name
+ * @param {String} value
+ * @return {String}
+ */
+function onIgnoreTagAttr (tag, name, value) {
+  // do nothing
+}
+
+/**
+ * HTML转义
+ *
+ * @param {String} html
+ */
+function escapeHtml (html) {
+  return html.replace(REGEXP_LT, '&lt;').replace(REGEXP_GT, '&gt;');
+}
+
+/**
+ * 安全的标签属性值
+ *
+ * @param {String} tag
+ * @param {String} name
+ * @param {String} value
+ * @param {Object} cssFilter
+ * @return {String}
+ */
+function safeAttrValue (tag, name, value, cssFilter) {
+  // 转换为友好的属性值，再做判断
+  value = friendlyAttrValue(value);
+
+  if (name === 'href' || name === 'src') {
+    // 过滤 href 和 src 属性
+    // 仅允许 http:// | https:// | mailto: | / | # 开头的地址
+    value = _.trim(value);
+    if (value === '#') return '#';
+    if (!(value.substr(0, 7) === 'http://' ||
+         value.substr(0, 8) === 'https://' ||
+         value.substr(0, 7) === 'mailto:' ||
+         value.substr(0, 4) === 'tel:' ||
+         value.substr(0, 11) === 'data:image/' ||
```

Comment:
I added this line and wanted to point out this change.  

We have a few image credits like the Cesium and Bing credits that use `<img src="data:image/png;base64,....."/>`.  `xss` was removing this.  There wasn't a way to whitelist it through the API so I added a line to allow it.  I couldn't find any reason why a data URI would lead to a security vulnerability, but I wanted to double check.

I added a comment to the top of the file so anyone updating this in the future knows that this file was altered from the original source.

Reply:
I would prefer we not modify this, both because it's third-party code and because even if we don't think there are security implications, there may be.  I don't know why we embed the Cesium and Bing credits anyway, they can just be external images like everything else.

If you are postive there are no security implications, then open an issue/PR in the xss respository and have it merged into the upstream project first.